### PR TITLE
feat(design-system): Add `Subtle` and `Link` variants to the `Button` XDS component.

### PR DESCRIPTION
**Pull request by tajespasarela**
EX-7508

Finally the `Subtle` variant will be named `Tight`

### CODE REVIEW

File: packages/x-tailwindcss/demo/src/components/xds-button.vue [hunk 1]
```diff
@@ -76,10 +82,14 @@
...
         Ghost: this.colors.map(addParentClasses(this.base, this.ghost)),
+        Tight: this.colors.map(addParentClasses(this.base, this.tight)),
+        Link: this.colors.map(addParentClasses(this.base, this.link)),
         Disabled: [
           this.base,
           addParentClasses(this.base)(this.outline),
-          addParentClasses(this.base)(this.ghost)
+          addParentClasses(this.base)(this.ghost),
+          addParentClasses(this.base)(this.tight),
+          addParentClasses(this.base)(this.link)
```
Comment (mnavarroespinosa) [inline packages/x-tailwindcss/demo/src/components/xds-button.vue:92]: What do you think of adding combinations including the new variants?

**Review by mnavarroespinosa — APPROVED** [review]

File: packages/x-tailwindcss/demo/src/components/xds-button.vue (hunk 1 above)
Comment (tajespasarela) [reply packages/x-tailwindcss/demo/src/components/xds-button.vue:94]: done

**Review by mnavarroespinosa — APPROVED** [review]

**Review by ajperezbau — CHANGES_REQUESTED** [review]
The cursor pointer seems to be `not-allowed` if the button is disabled.

Good job 👌

File: packages/x-tailwindcss/src/x-tailwind-plugin/components/button/link.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    link: {
+      paddingInlineStart: '0',
+      paddingInlineEnd: '0',
+      minHeight: '0',
+      textDecoration: 'underline',
+      fontFamily: 'inherit',
+      fontSize: 'inherit',
+      lineHeight: 'inherit',
+      fontWeight: 'inherit',
+      letterSpacing: 'inherit',
```
[the remark this answers is not in the document]
Comment (tajespasarela) [reply packages/x-tailwindcss/src/x-tailwind-plugin/components/button/link.ts:22]: done

**Comment by tajespasarela** [comment]
> The cursor seems to be `not-allowed` if the button is disabled.
> 
> Good job 👌

fixed

**Review by ajperezbau — APPROVED** [review]

**Pull request merged**